### PR TITLE
Fix misleading error messages

### PR DESCRIPTION
**Pull request by o-lim**
This fixes the case where the file, line number, and error message reported are incorrect, showing the error message for a previous call to `error()` that was caught with a `pcall()`.

```lua
describe('Test assert after pcall', function()
  it('should report line number of assert', function()
    local ok, msg = pcall(function() error('error message') end)
    assert(false, 'should report this line number')
  end)
end)
```

In the above example, the error message reported would be from the `pcall()`, instead of the error message from the `assert()`, where the actual test failure occurred.

The solution is to not replace `error` in the test environment. This can lead to more inconsistent error messages that may be difficult to trace.  Go back to rewriting the error message to insert the file and line
information for complex error objects.

### CODE REVIEW

File: spec/cl_spec.lua [hunk 1]
```diff
@@ -349,6 +349,24 @@ describe('Tests error messages through the command line', function()
     error_end()
   end)
 
+  it('when throwning an error table with __tostring', function()
```
Comment (DorianGray) [inline spec/cl_spec.lua]: typo!

File: spec/cl_spec.lua [hunk 2]
```diff
@@ -349,6 +349,24 @@ describe('Tests error messages through the command line', function()
...
+    error_start()
+    local result = run('bin/busted --output=plainTerminal --tags=tostring --pattern=cl_error_messages.lua$')
+    local errmsg = result:match('\n(%./spec/.-)\n')
+    local expected = './spec/cl_error_messages.lua:17: {}'
+    assert.is_equal(expected, errmsg)
+    error_end()
+  end)
+
+  it('when throwning after a pcall', function()
```
Comment (DorianGray) [inline spec/cl_spec.lua]: typo!

**Comment by DorianGray** [comment]
Typos aside, this looks really good. Will test and roll a new rc tonight.

**Comment by DorianGray** [comment]
I especially like moving nil handling into rewrite message, a much more robust approach.

**Comment by o-lim** [comment]
Just did a rebase to fix the typo.  Good catch.

**Comment by DorianGray** [comment]
I was able to test this early. Good to go!

**Pull request merged**